homeassistant/components/utility_meter/sensor.py

```diff
@@ -187,6 +194,7 @@ def __init__(
         self._sensor_net_consumption = net_consumption
         self._tariff = tariff
         self._tariff_entity = tariff_entity
+        self._attr_unique_id = f"utility_meter_{parent_meter}_{tariff}"
```

Comment:
This is based on the upstream entity ID. Entity IDs are not viable sources for unique IDs.

Reply:
parent_meter comes from the configuration slug of utility_meter (that's unique)
tariff is checked in the configuration to be unique

Reply:
I don't think that is right. For these kinds of helpers, we allow users to manually and explicitly define the unique ID as a configuration option. I think we should maintain that in this case.

Reply:
`parent_meter` comes from 
https://github.com/home-assistant/core/blob/f3b7d9565ec6cafa3407cbd69b2920a54e627996/homeassistant/components/utility_meter/sensor.py#L131

that line `meter` comes from https://github.com/home-assistant/core/blob/f3b7d9565ec6cafa3407cbd69b2920a54e627996/homeassistant/components/utility_meter/sensor.py#L109

CONF_METER is set in 2 places both carrying `meter`:
https://github.com/home-assistant/core/blob/f3b7d9565ec6cafa3407cbd69b2920a54e627996/homeassistant/components/utility_meter/__init__.py#L129
https://github.com/home-assistant/core/blob/f3b7d9565ec6cafa3407cbd69b2920a54e627996/homeassistant/components/utility_meter/__init__.py#L118

`meter` is the key in the configuration dict:
https://github.com/home-assistant/core/blob/f3b7d9565ec6cafa3407cbd69b2920a54e627996/homeassistant/components/utility_meter/__init__.py#L105

Reply:
Yep, and changing those configurations messes with the unique ID, thus not viable.

Please add a configuration parameter to allow configuring a unique ID, this will also make it consistent with other helpers that allow for this.

If anything, we should avoid different implementations to avoid confusion. Thanks 👍